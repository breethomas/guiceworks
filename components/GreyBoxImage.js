// @flow
import React from 'react'
import css, { media2 } from '../styles/css'
import View from './View'

type Props = {
  src: string,
}

const style = css(
  {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: '40px 20px',
    backgroundColor: '#f2f2f2',
  },
  media2({
    paddingTop: 120,
    paddingBottom: 120,
  }),
)

const imgStyle = css({
  width: '100%',
  maxWidth: 1170,
  marginRight: 'auto',
  marginLeft: 'auto',
})

const namify = src => src.split('.')[0].replace(/_|-/, ' ')

export default ({ src }: Props) => (
  <View className={style}>
    <img
      className={imgStyle}
      alt={namify(src)}
      src={src}
    />
  </View>
)

