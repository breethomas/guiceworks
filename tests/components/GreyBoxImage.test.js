import React from 'react'
import renderer from 'react-test-renderer'
import GreyBoxImage from '../../components/GreyBoxImage'

describe('GreyBoxImage', () => {
  let component

  beforeEach(() => {
    component = <GreyBoxImage src="AM_grey-box.jpg" />
  })

  it('shows the correct snapshot for the GreyBoxImage component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
