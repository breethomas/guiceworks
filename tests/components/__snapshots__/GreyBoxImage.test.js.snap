// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreyBoxImage shows the correct snapshot for the GreyBoxImage component tree 1`] = `
<div
  className={
    Object {
      "data-css-1vyftnr": "",
    }
  }
>
  <img
    alt="AM grey box"
    className={
      Object {
        "data-css-lsfykb": "",
      }
    }
    src="AM_grey-box.jpg"
  />
</div>
`;
